### ISSUE TITLE
Output formats

### ISSUE THREAD

**Issue by dbrgn** [dbrgn/miniaturo#4]
Should output image formats other than PNG be supported?

**Comment by dbrgn** [comment]
And if yes, how?

- File type could be determined from output filename (this might not be 100% API-compatible with raw-thumbnailer)
- File type could be determined using separate filetype parameter
- File type could be auto-determined if `--auto-type` is specified, otherwise fallback to PNG

I'm not sure what's best, but tend towards the first variant, even if it's not 100% compatible in edge cases (e.g. someone exporting to `thumb.jpg` and expecting it to be in PNG format anyways).

**Comment by dbrgn** [comment]
@hfiguiere what's your opinion on this?

**Comment by hfiguiere** [comment]
This depend of the thumbnail API but if it allows it I would output JPEG for case with the thumbnail is JPEG (no rencoding - save the bitstream directly) and PNG in the other cases.

But reading the Nautilus documentation, I don't think this is allowed.

**Comment by dbrgn** [comment]
Yeah, I would not change the behavior for the current default case. I think the caller of the API should be able to decide how to generate the thumbnails, either through the output filename or through a separate parameter.

**Comment by hfiguiere** [comment]
With PR#15 it will either output a JPEG or a PNG. The latter if the thumbnail is a pixmap, the former if it is already a JPEG. The libopenraw API doesn't have any other option.
